src/unplugin.ts

```diff
@@ -0,0 +1,101 @@
+import * as fs from "node:fs";
+import * as path from "node:path";
+import { log } from "console";
+import { createUnplugin } from "unplugin";
+import { maybeRewriteSourcemapWithReactProd, loadSourcemap } from "./index";
+
+export interface ReactSourcemapsPluginOptions {
```

Comment:
I wanted to also add a "preserve" option which would preserve the original sourcemaps with a original.js.map suffix, wdyt @markerikson ?

Reply:
Seems reasonable, sure!